AsyncDisplayKit/ASMultiplexImageNode.mm

```diff
@@ -456,8 +455,8 @@ - (void)_loadNextImage
     }];
   }
   // Likewise, if it's a iOS 8 Photo asset, we need to fetch it accordingly.
-  else if (AS_AT_LEAST_IOS8 && [[nextImageURL scheme] isEqualToString:kPHAssetURLScheme]) {
-    [self _loadPHAssetWithIdentifier:nextImageIdentifier URL:nextImageURL completion:^(UIImage *image, NSError *error) {
+  else if (ASPhotosFrameworkImageRequest *request = nextImageURL.asyncdisplaykit_photosRequest) {
```

Comment:
I realize that this pattern (.asyncdisplaykit_) is used in a few places, but I think those usages are usually quite necessary (like extensions to CALayer) without a clean way to avoid it.

In this case, is it not correct to say the flow is:
1. User creates a PhotosRequest object
2. User manually requests a URL from that object, and provides it to ASMultiplexImageNode
3. User discards reference to PhotosRequest
4. Internals of ASMultiplexImageNode revive the reference by grabbing associated object through magically-extended NSURL object

This is the only part of the diff that doesn't feel right to me.  Even parsing ph:// seems a bit better :).  Are there any real-world cases where Photos framework images would be mixed with regular remote URLs?  If not, what do you think about providing a load method that takes the ASPhotosFrameworkImageRequest directly?